include/nbl/builtin/hlsl/subgroup/arithmetic_portability_impl.hlsl

```diff
@@ -0,0 +1,490 @@
+// Copyright (C) 2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_SUBGROUP_ARITHMETIC_PORTABILITY_IMPL_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SUBGROUP_ARITHMETIC_PORTABILITY_IMPL_INCLUDED_
+
+#include "nbl/builtin/hlsl/binops.hlsl"
+#include "nbl/builtin/hlsl/subgroup/scratch.hlsl"
+#include "nbl/builtin/hlsl/subgroup/shuffle_portability.hlsl"
+
+// REVIEW:  Location and need of these. They need to be over a function but
+//          there's no need to have them over every subgroup func.
+//          The compiler doesn't seem to whine about these missing
+//          for subgroup*Min/Max functions even though the spec seems
+//          to mandate them https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#Group_Operation
+[[vk::ext_extension("GL_KHR_shader_subgroup_basic")]]
+[[vk::ext_extension("GL_KHR_shader_subgroup_arithmetic")]]
+[[vk::ext_capability(/* GroupNonUniformArithmetic */ 63)]]
+[[vk::ext_capability(/* GroupNonUniformBallot */ 64)]]
+void fake_for_capability_and_extension(){}
```

Comment:
put stuff like this in `impl` namespaces

`basic,vote,ballot,shuffle,shuffle_relative` extensions and caps can go in their corresponding (non-portability) headers

the rest can go in sections guarded by `NBL_` ifdefs!

Reply:
ok we did the godbolt tests (c.f. discord), you need to stack these on the original function intrinsic signatures that require the cap or extension, these `impl` functions are not reachable and don't result in emitting the caps and exts you intend to here